tests/std/tests/P0475R1_P0591R4_uses_allocator_construction/test.cpp

```diff
@@ -179,6 +179,18 @@ void test_GH_2021() { // COMPILE-ONLY
     tags[0];
 }
 
+struct MoveOnlyType {
+    MoveOnlyType()               = default;
+    MoveOnlyType(MoveOnlyType&&) = default;
+};
+
+void test_LWG3527() { // COMPILE-ONLY

```

Comment:
The tests are failing - it looks like this PR contains the test change, but not the product change, from #2194. Does this need to be merged with `main` now that #2194 is in?

Reply:
> The tests are failing - it looks like this PR contains the test change, but not the product change, from #2194. 

🤦 That is exactly the case. I was performing some verification of #2194 today and managed to leak this change across to this branch which I was squashing to PR. Fixing.